docs/content/how-to/restartannotation.md

```diff
@@ -0,0 +1,18 @@
+# Restart Control Plane Components with Restart Annotation
+
+The Restart Annotation allows a user to trigger a rolling restart of all control-plane components in the management cluster. This is helpful in certain situations like when you want to force a reload of all the certificates components are using in the control plane. It is also a helpful operational debug step to see if a restart resolves any abnormalities being experienced in the control plane components.
+
+To utilize: the restart annotation is passed to the Hostedcluster object
+
+```
+apiVersion: hypershift.openshift.io/v1alpha1
+kind: HostedCluster
+metadata:
+  name: example
+  namespace: master
+  annotations:
+    hypershift.openshift.io/restart-date: "2022-01-31T12:56:54+0000"
+....
+```
+
+When that annotation is passed: every component in the clusters control plane will proceed to do a rolling restart. This will occur anytime a new value is passed to the restart date annotation.
```

Comment:
Can we specify and enumerate all control-plane components?
kas
kcm
scheduler
openshift api server
Capi controller
capi provider controller
Ignition server

Anything else?

Reply:
I think this is clear enough and, if we listed them out, would be prone to getting out of date.  My $0.02.

Reply:
Provided a full list of everything that gets rolling restarted.

Reply:
Fine to remove the list as well whatever y'alls preference is

Reply:
>I think this is clear enough and, if we listed them out, would be prone to getting out of date. My $0.02.

Generally I would agree that listing adds little value here. However it's still not clear to me why all these components are subject to be restarted through an annotation.

The list is quite big atm and I think each component should be discussed and justified in a one per one basis while tracking issues for any underlying root problem which is being workarounded by restarting. E.g for this case https://github.com/openshift/hypershift/pull/941#discussion_r795863229 -> we should discuss how to approach cert reloading holistically.

Why are the autoscaler, the machine approver or the cluster version operator to name a few in this list?
If anything else is ever added to this annotation, there should be a discussion, an understood reason and possibly an issue tracking the underlying problem. I think meanwhile having these list documented helps to not loose track and raise visibility about the effective impact of this annotation, particularly useful for consumers with less context. @sjenning if you are ok with it I think there's value on having it.

Reply:
Ultimately as you operate a large scale offering: there needs to be tooling in place to do some basic operations like this independent of issue. Not saying it has to be utilized in every case but it needs to be available. There's a lot of situations internally that this has helped a lot on. Some for example:

- Recovering from a customer that has overloaded the kube-apiserver with requests and entered a extreme resource constrained state. It provides a way for them to fix the problem and then trigger a restart to recover. In an "ideal world" you would think kube recovers all on it's own but it doesn't in load conditions and needs a fresh restart at times to resync the state of the world and proceed forward.

- The more important one is CA rotation which is a requirement from a security perspective on our end otherwise you cannot fully revoke admin certs of the cluster. Go programs do not reload CA certs in their operations (they are loaded at startup and stay for the life of the program). These need to be restarted in order to pick up the CA certs.

There are more situations we can discuss as well but ultimately I believe this is a necessary operational requirement at scale and should be added to all components going forward. I would actually say we should not add a component that is part of the control plane without having this feature.

This might be good to talk on the sync up calls about. Thanks for everyone's time and reviews!

Reply:
We could consider if this is not wanted for AWS/etc just scoping it to IBM Cloud paths. But I would expect SRE teams operating offerings will ultimately desire this.